internal/versions/rest_api_versions.go

```diff
@@ -48,17 +48,16 @@ func (h *restAPIVersionsHandler) GetReleaseImage(_ context.Context, openshiftVer
 	if cpuArchitecture == common.MultiCPUArchitecture {
 		query = query.Where("cardinality(cpu_architectures) > 1")
 	} else {
-		query = query.Where(h.db.Session(&gorm.Session{}).Where("? = ANY(cpu_architectures)", cpuArchitecture).Or("cpu_architecture = ?", cpuArchitecture))
+		query = query.Where(h.db.Session(&gorm.Session{}).
+			Where("? = ANY(cpu_architectures) AND cardinality(cpu_architectures) = 1", cpuArchitecture).
+			Or("cpu_architecture = ?", cpuArchitecture))
 	}
 
 	// Find the exact version
 	if versionFormat == common.MajorMinorPatchVersion {
-		editedVersion := openshiftVersion
-		// add multi suffix if missing
-		if cpuArchitecture == common.MultiCPUArchitecture && !strings.HasSuffix(openshiftVersion, "-multi") {
-			editedVersion = editedVersion + "-multi"
-		}
-		query = query.Where("version = ?", editedVersion)
+		query = query.Where(h.db.Session(&gorm.Session{}).
+			Where("version = ?", openshiftVersion).
+			Or("version = ?", openshiftVersion+"-multi"))
```

Comment:
Why change this?
Shouldn't we continue only searching for multi-arch images if multi was specifically requested?

Reply:
There are some versions given in a different format even in restAPI (like in ABI case) where `multi` is not in the suffix